tensorflow/compiler/tf2tensorrt/utils/trt_tensor_proxy.h

```diff
@@ -0,0 +1,599 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_COMPILER_TF2TENSORRT_CONVERT_TRT_TENSOR_PROXY_H
+#define TENSORFLOW_COMPILER_TF2TENSORRT_CONVERT_TRT_TENSOR_PROXY_H
+
+#include <vector>
+#include <string>
+#include <cassert>
+#include <memory>
+
+#if GOOGLE_CUDA && GOOGLE_TENSORRT
+#include "third_party/tensorrt/NvInfer.h"
+namespace tensorflow {
+
+namespace tensorrt {
+
+namespace {
+  nvinfer1::Dims GetTestDims(const std::vector<int>& d) {
+    nvinfer1::Dims dims;
+    dims.nbDims = d.size();
+    for (int i = 0; i < d.size(); ++i) {
+      dims.d[i] = d[i];
+    }
+  }
+}
+// A fake ITensor implementation used to check whether the TF-TRT converter can
+// handle specific node. We only need shape and type information, and the
+// converter won't (and shouldn't) use this to build the TRT network.
+class SimpleITensor {
+ public:
+  SimpleITensor(nvinfer1::DataType trt_dtype, const nvinfer1::Dims& trt_dims)
+      : trt_dtype_(trt_dtype), trt_dims_(trt_dims) {}
+
+  void setName(const char* name) {}
+
+  const char* getName() const { return ""; }
+
+  void setDimensions(nvinfer1::Dims dimensions) {
+    trt_dims_ = dimensions;
+  }
+
+  nvinfer1::Dims getDimensions() const { return trt_dims_; }
+
+  void setType(nvinfer1::DataType trt_dtype) {
+    trt_dtype_ = trt_dtype;
+  }
+
+  nvinfer1::DataType getType() const { return trt_dtype_; }
+
+  bool isNetworkInput() const { return false; }
+
+  bool isNetworkOutput() const { return false; }
+
+  void setBroadcastAcrossBatch(bool broadcastAcrossBatch) {}
+
+  bool getBroadcastAcrossBatch() const { return false; }
+
+  nvinfer1::TensorLocation getLocation() const { return location_; }
+
+  void setLocation(nvinfer1::TensorLocation location) {
+    location_ = location;
+  }
+  bool setDynamicRange(float min, float max) {
+    dynamic_range_ = std::max(std::abs(min), std::abs(max));
+    return true;
+  }
+  
+  float getDynamicRange() const { return dynamic_range_; }
+  bool dynamicRangeIsSet() const { return true; }
+
+  void resetDynamicRange() {}
+
+  float getDynamicRangeMin() const { return 0.f; }
+
+  float getDynamicRangeMax() const { return 0.f; }
+
+  void setAllowedFormats(nvinfer1::TensorFormats formats) {}
+
+  nvinfer1::TensorFormats getAllowedFormats() const { return 1; }
+
+  bool isShapeTensor() const { return false; }
+  bool isExecutionTensor() const { return true; }
+
+ private:
+  nvinfer1::DataType trt_dtype_;
+  nvinfer1::Dims trt_dims_;
+  std::string name_;
+  nvinfer1::TensorLocation location_;
+  float dynamic_range_;
+};
+
+// Fake ITensor implementation for testing purposes.
+class FakeITensor {
```

Comment:
This class is the same as SimpleITensor, is it? Maybe should remove this class and use SimpleITensor in the test?

Reply:
Good point; merged the two classes in the latest commit, thanks